diff --git a/password_policy_history/password_policy_history.module b/password_policy_history/password_policy_history.module
index 2237100..24d859d 100644
--- a/password_policy_history/password_policy_history.module
+++ b/password_policy_history/password_policy_history.module
@@ -29,20 +29,25 @@ function password_policy_history_user_update(AccountInterface $account) {
  *   The current user account.
  */
 function _password_policy_history_insert_password_hash(AccountInterface $account) {
-  if (empty($_POST['pass']['pass1']) || empty($_POST['pass']['pass2'])) {
-    return;
-  }
-  $connection = Database::getConnection();
+  // Somehow this hook gets called twice, resulting in the password being added
+  // twice.  This makes sure it only gets inserted once.
+  if(!isset($account->savedPassHistory)) {
+    if (empty($_POST['pass']['pass1']) || empty($_POST['pass']['pass2'])) {
+      return;
+    }
+    $connection = Database::getConnection();
 
-  $hashed_pass = $account->getPassword();
-  $values = [
-    $account->id(),
-    $hashed_pass,
-    \Drupal::time()->getRequestTime(),
-  ];
-  $connection->insert('password_policy_history')
-    ->fields(['uid', 'pass_hash', 'timestamp'], $values)
-    ->execute();
+    $hashed_pass = $account->getPassword();
+    $values = [
+      $account->id(),
+      $hashed_pass,
+      \Drupal::time()->getRequestTime(),
+    ];
+    $connection->insert('password_policy_history')
+      ->fields(['uid', 'pass_hash', 'timestamp'], $values)
+      ->execute();
+    $account->savedPassHistory = TRUE;
+  }
 }
 
 /**
